assets/js/modules/reader-revenue-manager/datastore/publications.js

```diff
@@ -94,6 +190,7 @@ const baseResolvers = {
 			.getPublications();
 		if ( publications === undefined ) {
 			yield fetchGetPublicationsStore.actions.fetchGetPublications();
+			yield baseActions.syncPublicationOnboardingState();
```

Comment:
@techanvil @ankitrox Why did we add this here? Syncing the onboarding state only makes sense in the context of a selected publication in settings which this selector/resolver is not related to. Also, since this will usually be called before any selection is made I would expect this to not have the intended result. We should only sync the onboarding state for the publication which is saved in settings, not simply selected in state.

Reply:
@aaemnnosttv this is defined as a requirement in the [design doc](https://docs.google.com/document/d/14CxsQjTpdPrQpd8NoQtXnUd1DG7FVtmF5GTVvYdixVY/edit#bookmark=id.9bqgtkjartaf), and the AC/IB for this issue. So, I haven't really questioned it. My presumption is that the intention is to resync the onboarding state for the currently selected publication any time we visit the Settings screen after it has been connected. Arguably, though, it could make sense to decouple the sync from `getPublications()`. Maybe @nfmohit can shed a bit more light on this one?

Reply:
Thank you for raising this, @aaemnnosttv. @techanvil is correct with the presumption above. Since we show the `PublicationOnboardingStateNotice` in the settings screen based on the `publicationOnboardingState` module setting which is updated once every hour, there is a chance we may show an unintended notice due to the onboarding state not being up to date.

Since the `getPublications` selector would likely be resolved in `SettingsEdit`, my intention was to dispatch the `syncPublicationOnboardingState` action that would update the onboarding state in the module settings. The action only does this if the module is connected and a publication ID is set.

This would ensure that the correct `PublicationOnboardingStateNotice` is shown (or not shown at all). Would you suggest we do this any differently?

Reply:
Now that I think deeply about this, I feel this is not really necessary and does not add much value and has the potential to cause problems as Evan mentioned above. @techanvil What do you think of the idea of removing this dispatch as part of a subsequent issue?

Reply:
If the intention is to sync the state on the settings view, then I think it would be more appropriate to move the effect to that component as it's not related to this selector.

Reply:
Thanks @nfmohit and @aaemnnosttv. Agreed, it would be good to move the sync out to the component.

I think this merits a new issue to flesh out the details. We also need to revisit the `PublicationSelect` component due to the divergence from the IB in PR https://github.com/google/site-kit-wp/pull/9006:

> Note: `publicationOnboardingStateLastSyncedAtMs` is not getting updated as part of `onPublicationChange` function because it will be updated as part of the `syncPublicationOnboardingState` action, which will run immediately before (`getPublications` resolver)/after (`maybeSyncPublicationsOnboardingState`) a publication is selected.

@nfmohit, would you be happy to create an issue for this?

Reply:
> Agreed, it would be good to move the sync out to the component.
> 
> I think this merits a new issue to flesh out the details.

Thank you @techanvil. About this one, now that I think about it, I feel updating it in `SettingsEdit` might actually have unexpected results. The action is responsible for showing the publication approved overlay notification by changing a `core/ui` data store key, which would only happen when the action is dispatched in the main dashboard. Also, the state would be updated once every hour anyway, so just updating it only for `SettingsEdit` doesn't seem to be worth it while the publication onboarding state notice also appears in `SettingsView`.

> I think this merits a new issue to flesh out the details. We also need to revisit the `PublicationSelect` component due to the divergence from the IB in PR #9006:
> 
> > Note: `publicationOnboardingStateLastSyncedAtMs` is not getting updated as part of `onPublicationChange` function because it will be updated as part of the `syncPublicationOnboardingState` action, which will run immediately before (`getPublications` resolver)/after (`maybeSyncPublicationsOnboardingState`) a publication is selected.

About this one, I also feel updating `publicationOnboardingStateLastSyncedAtMs` as part of `syncPublicationOnboardingState` is sufficient.

Please let me know what you think, thanks so much!

Reply:
> Thank you @techanvil. About this one, now that I think about it, I feel updating it in `SettingsEdit` might actually have unexpected results. The action is responsible for showing the publication approved overlay notification by changing a `core/ui` data store key, which would only happen when the action is dispatched in the main dashboard. Also, the state would be updated once every hour anyway, so just updating it only for `SettingsEdit` doesn't seem to be worth it while the publication onboarding state notice also appears in `SettingsView`. To be fair, dispatching this as a part of the resolver was planned for redundancy.

Thanks @nfmohit. As you've pointed out, the publication onboarding state notice appears both in `SettingsEdit` and `SettingsView`, so it's a bit of an oddity to only sync the state in `SettingsEdit`. I think we could either dispatch the sync in `SettingsView` instead, ensuring the notice is up-to-date in both places, or not at all, and allow for the fact it could be stale. Obviously from UX perspective it would be better if it was up-to-date, so I'd lean in that direction; that said we might then want to consider a loading state which could add a bit of complexity. Maybe we can simply omit it for now and add it as an improvement post launch.

If we do still sync on the Settings screen, setting the `core/ui` value won't have any effect so I don't see it being too much of a concern here; that said we could still amend the action so as to avoid that if not on the dashboard.

> > > Note: `publicationOnboardingStateLastSyncedAtMs` is not getting updated as part of `onPublicationChange` function because it will be updated as part of the `syncPublicationOnboardingState` action, which will run immediately before (`getPublications` resolver)/after (`maybeSyncPublicationsOnboardingState`) a publication is selected.
> 
> About this one, I also feel updating `publicationOnboardingStateLastSyncedAtMs` as part of `syncPublicationOnboardingState` is sufficient.

Hmm. If we are removing the call to `syncPublicationOnboardingState()` from the Settings screen entirely, then the logic about it being run immediately before a publication is selected no longer holds true.

So, we could find ourselves making another call to the API in order to sync the onboarding state when we land on the dashboard, even though we've just updated the value.

I would say it would be preferable to avoid this to help reduce unneeded API requests and the relevant API quota usage. So my preference would be to update the timestamp if we are not syncing on the Settings screen. However, it's a pretty marginal point and again something we could address post launch.

Reply:
Thank you for your input, Tom! I've included updating this module setting as part of my `selectPublication` action proposed in [my PR](https://github.com/google/site-kit-wp/pull/9098) for [#8800](https://github.com/google/site-kit-wp/issues/8800).

Reply:
Looks good, nice one Nahid!